### DISCUSSION TITLE
How to check if the run is the first run or a re-processing of the same period

### DISCUSSION THREAD

**Discussion by timothy-khom** [apache/airflow#34662] [Q&A]
Hi colleagues,

We have implemented listener mechanism to react on started, succeeded, failed dags/tasks events according to https://airflow.apache.org/docs/apache-airflow/stable/_modules/airflow/example_dags/plugins/event_listener.html.

On raising dag running event we need to figure out if this is the first run for this execution date (data_interval_start) or this is the re-processing of the same period. 
There is DagRun is_backfill property, but based on the source code it is related only to Backfill job.
We are interested in any re-run, e.g. when the run is clearing from Airflow UI.

The idea we tried was to check if the are previous successful runs for the same date period by querying dag_run table:
```
previous_runs_count: List[DagRun] = session.query(DagRun).filter(
        DagRun.dag_id == dag.dag_id,
        DagRun.data_interval_start == data_interval_start,
        DagRun.data_interval_end == data_interval_end,
        DagRun.state == "success",
        DagRun.start_date < dag_run.start_date
    ).count()
```

 But it turned out that at the time **on_dag_run_running** event is fired there is always the only record with running status for the required data interval. Basically there is always the single record for any dag_run, which is just updated.

Could you help find a way to implement the logic we need? Maybe there is a table where all the dag_runs history is stored?

**Comment by Taragolis** [comment]
Airflow doesn't store historical information for DagRun
